### PR TITLE
fix: do not show test explorer related error for client which don't implement it

### LINKED ISSUE #3660: Test Explorer won't work... message shouldn't be shown in clients that don't implement the test explorer
### Describe the bug

When using a client that doesn't implement the test explorer and Metals detects something that might now work right you get a warning that pops up. For example when using `nvim-metals` I hit on this today:

```
Test Explorer won't work due to mis-configuration. Select 'More information' to learn how to fix this problem..

Request Actions:
1. More information
2. Don't show again
Type number and <Enter> or click with the mouse (q or empty cancels):
```

### Expected behavior

`nvim-metals` doesn't implement the test explorer so any issues related to it shouldn't surface to the client. This is probably due to the problem resolver in metals looking for things like outdated junit interfaces even in situations where the explorer won't be used. There should probably be extra checks there.

### Operating system

macOS

### Editor/Extension

Nvim (nvim-metals)

### Version of Metals

0.11.1+161-ff365f7c-SNAPSHOT

### Extra context or search terms

_No response_

### PR DESCRIPTION
**Pull request by kpodsiad**
fixes #3660

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/troubleshoot/ProblemResolver.scala [hunk 1]
```diff
@@ -19,7 +19,8 @@ class ProblemResolver(
     workspace: AbsolutePath,
     mtagsResolver: MtagsResolver,
     currentBuildServer: () => Option[BspSession],
-    javaHome: () => Option[String]
+    javaHome: () => Option[String],
+    isTestExplorerProvider: () => Boolean
```
Comment (kpodsiad) [inline metals/src/main/scala/scala/meta/internal/troubleshoot/ProblemResolver.scala:23]: I'm not a fan of passing boolean flags, but it was easier than passing whole `clientConfig`

File: metals/src/main/scala/scala/meta/internal/metals/Doctor.scala [hunk 2]
```diff
@@ -45,7 +45,8 @@ final class Doctor(
       workspace,
       mtagsResolver,
       currentBuildServer,
-      javaHome
+      javaHome,
+      () => clientConfig.isCompletionItemResolve()
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/Doctor.scala:49]: This seems like the wrong config, no?

Comment (kpodsiad) [reply metals/src/main/scala/scala/meta/internal/metals/Doctor.scala:49]: wow, I don't have any idea why I did this 😳

**Review by ckipp01 — APPROVED** [review]
👍🏼 thanks for fixing this!

**Review by tgodzik — APPROVED** [review]
LGTM!

**Comment by kpodsiad** [comment]
One test on windows is failing but it's not related to my changes so I'm merging.

**Pull request merged**